tests/framework/clients/k3d/k3d.go

```diff
@@ -0,0 +1,160 @@
+		`--k3s-server-arg=--no-deploy=servicelb`,
+		`--k3s-server-arg=--no-deploy=metrics-serve`,
+		`--k3s-server-arg=--no-deploy=local-storage`).CombinedOutput()
+	if err != nil {
+		return nil, errors.Wrap(err, "CreateK3DCluster: "+string(msg))
+	}
+
+	configBytes, err := exec.Command("k3d", "kubeconfig", "get", name).Output()
+	if err != nil {
+		return nil, errors.Wrap(err, "CreateK3DCluster: failed to get kubeconfig for k3d cluster")
+	}
+
+	restConfig, err := clientcmd.RESTConfigFromKubeConfig(configBytes)
+	if err != nil {
+		return nil, errors.Wrap(err, "CreateK3DCluster: failed to parse kubeconfig for k3d cluster")
+	}
+
+	return restConfig, nil
+}
+
+// DeleteK3DCluster deletes the k3d cluster with the given name. An error is returned if the cluster does not exist.
+func DeleteK3DCluster(name string) error {
+	return exec.Command("k3d", "cluster", "delete", name).Run()
+}
+
+// CreateAndImportK3DCluster creates a new k3d cluster and imports it into rancher.
+func CreateAndImportK3DCluster(client *rancher.Client, name string) (*management.Cluster, error) {
+	var err error
+
+	name = defaultName(name)
+
+	// create the provisioning cluster
+	cluster := &apisV1.Cluster{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: "fleet-default",
+		},
+	}
+	_, err = client.Provisioning.Clusters("fleet-default").Create(context.TODO(), cluster, metav1.CreateOptions{})
+	if err != nil {
+		return nil, errors.Wrap(err, "CreateAndImportK3DCluster: failed to create provisioning cluster")
+	}
+
+	// create the k3s cluster
+	downRest, err := CreateK3DCluster(client.Session, name)
+	if err != nil {
+		return nil, errors.Wrap(err, "CreateAndImportK3DCluster: failed to create k3d cluster")
+	}
+
+	// wait for the management cluster
+	mClusterWatch, err := client.GetManagementWatchInterface(management.ClusterType, metav1.ListOptions{})
+	if err != nil {
+		return nil, errors.Wrap(err, "CreateAndImportK3DCluster: failed to watch for management cluster")
+	}
+
+	var mgmtCluster *management.Cluster
+	err = wait.WatchWait(mClusterWatch, func(event watch.Event) (bool, error) {
+		var mc v3.Cluster
+		err = runtime.DefaultUnstructuredConverter.FromUnstructured(event.Object.(*unstructured.Unstructured).Object, &mc)
+		if mc.Spec.DisplayName == name {
+			mgmtCluster, err = client.Management.Cluster.ByID(mc.Name)
+			return true, err
+		}
+
+		return false, nil
+
+	})
+	if err != nil {
+		return nil, errors.Wrap(err, "CreateAndImportK3DCluster: failed to watch for management cluster")
+	}
+
+	// import the k3d cluster
+	err = clusters.ImportCluster(client, mgmtCluster, downRest)
+	if err != nil {
+		return nil, errors.Wrap(err, "CreateAndImportK3DCluster: failed to import cluster")
+	}
+
+	// wait for cluster to be ready
+	mClusterWatch, err = client.GetManagementWatchInterface(management.ClusterType, metav1.ListOptions{})
+	if err != nil {
+		return nil, errors.Wrap(err, "CreateAndImportK3DCluster: failed to wait for management cluster ready")
+	}
```

Comment:
because this is a different cluster. Provisioning has 1 cluster type that leads to a management cluster being created. I need to get the management cluster to get the import tokens but I have to start by creating the provisioning cluster.